__tests__/integration/utils/wallet-helper.js

```diff
@@ -146,18 +155,42 @@ export class WalletHelper {
       walletsArr.unshift(new WalletHelper(genesis.walletId, { words: genesis.words }));
     }
 
-    // Requests the start of all the wallets in quick succession
-    const startPromisesArray = [];
-    for (const wallet of walletsArr) {
-      const promise = TestUtils.startWallet(wallet.walletData);
-      walletsPendingReady[wallet.walletId] = wallet;
-      startPromisesArray.push(promise);
+    // Start of the requests
+    startBenchmark.requestsStart = Date.now().valueOf();
+
+    if (options.serial || walletsArr.length > 2) {
+      // If we need to initialize too many wallets at once, it's better to do it serially
+      for (const wallet of walletsArr) {
+        const walletBenchmark = {};
+        walletBenchmark.requestStart = Date.now().valueOf();
+        await TestUtils.startWallet(wallet.walletData, {
+          waitWalletReady: true
+        });
+        walletBenchmark.requestEnd = Date.now().valueOf();
+        walletsPendingReady[wallet.walletId] = wallet;
+        walletBenchmark.diffRequest = walletBenchmark.requestEnd
+          - walletBenchmark.requestStart;
+        startBenchmark.wallets[wallet.walletId] = walletBenchmark;
+      }
+    } else {
+      // Requests the start of all the wallets in quick succession - parallel mode
+      const startPromisesArray = [];
+      for (const wallet of walletsArr) {
+        const promise = TestUtils.startWallet(wallet.walletData);
+        walletsPendingReady[wallet.walletId] = wallet;
+        const walletBenchmark = {};
+        walletBenchmark.requestStart = Date.now().valueOf();
+        startBenchmark.wallets[wallet.walletId] = walletBenchmark;
+        startPromisesArray.push(promise);
+      }
+      await Promise.all(startPromisesArray);
     }
-    await Promise.all(startPromisesArray);
+
+    startBenchmark.requestsEnd = Date.now().valueOf();
+    startBenchmark.requestsDiff = startBenchmark.requestsEnd - startBenchmark.requestsStart;
 
     // Enters the loop checking each wallet for its status
-    const timestampStart = Date.now().valueOf();
-    const timestampTimeout = timestampStart + testConfig.walletStartTimeout;
+    const timestampTimeout = startBenchmark.requestsEnd + testConfig.walletStartTimeout;
```

Comment:
The idea here is to limit the time the second phase ( `/status` double-checking loop ) can take.

Using The `requestsStart` ( renamed to `startRequestsBegin` on [7c199cd](https://github.com/HathorNetwork/hathor-wallet-headless/pull/164/commits/7c199cdfff0a5892ea114cd2726dd2ce37b90318) ) would mix it with the time of the first phase too.

Could you check if the variable renaming made this intention clearer?

Reply:
So the timeout is not for the wallet start, it's for the wallet ready?

Reply:
You are right, since we are using the `walletStartTimeout`, the best time reference for the timeout would be the `startRequestsBegin`. Fixed this on [a9225bb](https://github.com/HathorNetwork/hathor-wallet-headless/pull/164/commits/a9225bb4fa7ea5fbeb355e4f50a5ed11aa8b70f9)

In practice, we only have this _timeout_ check to avoid a permanent wait on a connection to the fullnode that may be broken and not recoverable. The delay between `startRequestsBegin` and `startRequestsEnd` is usually minimal compared to the loop check time.